torchx/lib/torchx/backend.ex

```diff
@@ -748,6 +748,48 @@ defmodule Torchx.Backend do
     |> to_nx(out)
   end
 
+  # shape: shape, type: type
+  @impl true
+  def select(out, %{shape: {}} = pred, on_true, on_false) do
+    if to_number(pred) == 0 do
+      on_false
+      |> from_nx()
+      |> Torchx.broadcast_to(out.shape)
+      |> to_nx(out)
+    else
+      on_true
+      |> from_nx()
+      |> Torchx.broadcast_to(out.shape)
+      |> to_nx(out)
+    end
+  end
+
+  def select(%T{type: out_type} = out, pred, on_true, on_false) do
+    on_true_torch =
+      on_true
+      |> from_nx()
+
+    on_false_torch =
+      on_false
+      |> from_nx()
+
+    if out_type == {:u, 8} do
+      pred_torch =
+        pred
+        |> from_nx()
+
+      to_nx(Torchx.where(pred_torch, on_true_torch, on_false_torch), out)
+    else
+      invert_pred_torch =
+        pred
+        |> from_nx()
+        |> Torchx.logical_not()
+
+      # swap true/false tensor
+      to_nx(Torchx.where(invert_pred_torch, on_false_torch, on_true_torch), out)
```

Comment:
I didn't quite get this if/else. Could you explain?

Reply:
We need to convert stuff to boolean and logical_not does it. But we need to swap the predicates!

Reply:
Perhaps this could be a comment in the code